### PR TITLE
fix: token and cookie session id miss match.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

related feature not in use yet, so the pr is safe.
find something need improvement when working with the clients.

1.  session token use the session id generated when processing cookie and stored in context
2.  small refactor of request login and refresh  

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16786)

### CODE REVIEW

**Comment by youngsofun** [comment]
need to wait for https://github.com/databendlabs/databend/pull/16786 to merge

File: tests/sqllogictests/suites/temp_table/alter_temp_tables.sql [hunk 1]
```diff
@@ -193,8 +193,8 @@ alter table t modify c1 varchar comment 'c1-column', c2 int comment 'test';
 query TTTTT
 select database,table,name,data_type,comment from system.columns where table='t' and database='default';
 ----
-default t c1 VARCHAR 'c1-column'
-default t c2 INT 'test'
+default t c1 VARCHAR c1-column
```
Comment (sundy-li) [inline tests/sqllogictests/suites/temp_table/alter_temp_tables.sql:196]: which codes lead these changes?

Comment (youngsofun) [reply tests/sqllogictests/suites/temp_table/alter_temp_tables.sql:196]: no idea too

Comment (youngsofun) [reply tests/sqllogictests/suites/temp_table/alter_temp_tables.sql:196]: but the new result is correct

**Review by sundy-li — APPROVED** [review]

**Pull request merged**